sim/dex-data.ts

```diff
@@ -905,70 +911,72 @@ export class Move extends BasicEffect implements Readonly<BasicEffect & MoveData
 		this.stab = data.stab || undefined;
 		this.volatileStatus = typeof data.volatileStatus === 'string' ? (data.volatileStatus as ID) : undefined;
 
-		if (this.category !== 'Status' && !this.gmaxPower) {
+		if (this.category !== 'Status' && !this.maxMove?.basePower) {
+			this.maxMove = this.maxMove || {} as MoveData['maxMove'];
 			if (this.isMax || this.isZ) {
-				this.gmaxPower = 1;
+				this.maxMove!.basePower = 1;
 			} else if (!this.basePower) {
-				this.gmaxPower = 100;
+				this.maxMove!.basePower = 100;
 			} else if (['Fighting', 'Poison'].includes(this.type)) {
 				if (this.basePower >= 150) {
-					this.gmaxPower = 100;
+					this.maxMove!.basePower = 100;
 				} else if (this.basePower >= 110) {
-					this.gmaxPower = 95;
+					this.maxMove!.basePower = 95;
 				} else if (this.basePower >= 75) {
-					this.gmaxPower = 90;
+					this.maxMove!.basePower = 90;
 				} else if (this.basePower >= 65) {
-					this.gmaxPower = 85;
+					this.maxMove!.basePower = 85;
 				} else if (this.basePower >= 55) {
-					this.gmaxPower = 80;
+					this.maxMove!.basePower = 80;
 				} else if (this.basePower >= 45) {
-					this.gmaxPower = 75;
+					this.maxMove!.basePower = 75;
 				} else {
-					this.gmaxPower = 70;
+					this.maxMove!.basePower = 70;
 				}
 			} else {
 				if (this.basePower >= 150) {
-					this.gmaxPower = 150;
+					this.maxMove!.basePower = 150;
 				} else if (this.basePower >= 110) {
-					this.gmaxPower = 140;
+					this.maxMove!.basePower = 140;
 				} else if (this.basePower >= 75) {
-					this.gmaxPower = 130;
+					this.maxMove!.basePower = 130;
 				} else if (this.basePower >= 65) {
-					this.gmaxPower = 120;
+					this.maxMove!.basePower = 120;
 				} else if (this.basePower >= 55) {
-					this.gmaxPower = 110;
+					this.maxMove!.basePower = 110;
 				} else if (this.basePower >= 45) {
-					this.gmaxPower = 100;
+					this.maxMove!.basePower = 100;
 				} else {
-					this.gmaxPower = 90;
+					this.maxMove!.basePower = 90;
 				}
 			}
 		}
-		if (this.category !== 'Status' && !this.zMovePower && !this.isZ && !this.isMax) {
+		if (this.category !== 'Status' && !this.zMove?.basePower && !this.isZ && !this.isMax) {
```

Comment:
```suggestion
		if (this.category !== 'Status' && !this.zMove && !this.isZ && !this.isMax) {
```

Reply:
Isn't this incorrect? I guess I dont know how zmoves work, but do not some have both a boost/effect and base power?

Reply:
There are unique z-moves that have both, but they're unique z-moves with their own entries in the Moves table. `move.zMove` is about generic z-moves, which for status moves are a `boost`/`effect` on top of their existing effect, and for physical/special moves are a high-damage (defined by `basePower`) Protect-breaking move with no other effect.

If a move already has generic Z-move data defined, this function should not overwrite or supplement it.